test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,633 @@
+
+        # Now retrieve all chats using the correct chat_id
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id,  # Use the chat_id from the initial chat creation
+            "page": 1,
+            "page_size": 1000
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        # Check for successful response
+        self.assertEqual(response.status_code, 200, "Failed to retrieve chats.")
+
+        response_data = response.json()
+
+        # Check that 'messages' is a list
+        messages = response_data.get('messages')
+        self.assertIsInstance(messages, list, "Expected 'messages' to be a list.")
+
+        # Verify that all added messages are in the response
+        for expected_message in messages_to_add:
+            self.assertTrue(any(msg.get('message') == expected_message for msg in messages),
+                            f"Expected message '{expected_message}' not found in the response.")
+
+        logger.info("=== Test Case 11 Completed ===\n")
+
+    def test_high_page_number(self):
+        """Test case 12: High Page Number"""
+        logger.info("=== Starting Test case 12: High Page Number ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,  # Valid customer GUID
+            "chat_id": self.valid_chat_id,  # Valid chat ID
+            "page": 1000,  # High page number
+            "page_size": 10  # Valid page size
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        # Check that the status code is 404
+        self.assertEqual(response.status_code, 404, "Expected status code 404 for high page number.")
+
+        # Verify that the response text contains the expected error message
+        self.assertIn("no chats found for this customer and chat id", response.text.lower(),
+                      "Expected error message not found in the response.")
+        logger.info("=== Test case 12: High Page Number Completed ===\n")
+
+    def test_retrieve_with_large_page_size(self):
+        """Test case 13: Retrieve 20 chats when only 5 exist"""
+        logger.info("=== Test Case 13 : Retrieve 20 chats when only 5 exist ===")
+
+        # Add 5 chat messages to the chat
+        chat_url = f"{self.BASE_URL}/chat"
+        messages_to_add = [
+            "Message 1: You will get the correct answer once AI is integrated.",
```

Comment:
These all should be questions

Reply:
I changed the question in this testcase, sir.